### PR TITLE
Resolve #782 -- ClientID corrected

### LINKED ISSUE #782: Menu transfer bug
Sometimes when 2 players connect at the same time, one gets the menu of the other, so they can't cast their spells, also they can't levelup spells, seems like race conditions with packets, and wrong sending.

e.g. Ez and Lulu connect at the same time, Lulu gets Ez menu (spells, etc), and Ez gets Lulu menu, and the menu doesn't work, also chat seems to be inversed.

### PR DESCRIPTION
**Pull request by danil179**
* Fixed the whole system of PlayerId and ClientId
* Changed PlayerId (also called UserId) to be ulong
* Added ClientId to be used where it needed

Refs #782 #778

Also that's some Progress #778 

Tested with 2 players localhost (order of connection now fixed, so if player 2 connect first it will get the correct menu and not the other player menu)

Why so many files got changed? That's cause there is problems with using int for ulong so I tried to fix it in some places. Still many refactoring needed, but I don't want to change EVERY file.

I added tip so players will know what champion the server thinks they play.

### CODE REVIEW

**Comment by lzardy** [comment]
Does not fix the issue. Situation where issue will occur: Have 3 players, join as P1, join as P3, wait until the "spawning map" line is printed twice, then join as P2. P1 screen will show P3 screen without actually being able to control P3s character. Edit: You will also be able to control P1, but it will be limited as it still thinks you are P3.

**Comment by danil179** [comment]
@microsoftv Fixed this problem. Now it should work. The problem was that each hero should be informed about other HeroSpawn once, but the old implementation informed every champion about the connection of the new champion, so it overrided the first client whenever there was unconnected client (cause the unconnected one send ClientId=0).

I fixed this by do all the informing just before game start and not on spawn packets (which you get for each player connection)

**Comment by lzardy** [comment]
Working as intended, 5v5 games are now seemingly stable connection wise.

File: GameServerCore/NetInfo/ClientInfo.cs [hunk 1]
```diff
@@ -5,10 +5,11 @@ namespace GameServerCore.NetInfo
 {
     public class ClientInfo
     {
-        public long UserId { get; private set; }
-        public int PlayerNo { get; set; }
+        public ulong PlayerId { get; private set; }
+        public uint ClientId { get; set; }
         public bool IsMatchingVersion { get; set; }
         public bool IsDisconnected { get; set; }
+        public bool IsStartedClient { get; set; }
```
Comment (Veykril) [inline GameServerCore/NetInfo/ClientInfo.cs:12]: Personally I would call it `IsClientStarted`, but that's just my personal preference. Looks all good to me at least.

**Review by Veykril — APPROVED** [review]

**Pull request merged**